### DISCUSSION TITLE
Commit messages

### DISCUSSION THREAD

**Discussion by jondequinor** [equinor/ert#2095] [General]
`CONTRIBUTING.md` does not constrain or guide us in how to write commit messages. Should it?

In the team primarily developing ERT, we have, internally, the following constraints for writing commits:

- a commit should do one atomic change on the repository and
- the commit message should be descriptive.

There's also a reference to https://chris.beams.io/posts/git-commit/

Here I propose that we include all of this to `CONTRIBUTING.md`, but I want to add another point:

> Just focus on making clear the reasons **why** you made the change in the first place—the way things worked before the change (and what was wrong with that), the way they work now, and why you decided to solve it the way you did.

I want to include this, because I am currently having a hard time reading the ERT commit history. It seems to be very _how_ focused. The code is in itself answering the _how_.

What do you think?

**Comment by sondreso** [comment]
I think this is a very good idea! I would also vote for including a sentence stating that a commit body is also preferred, I think this is underused at the moment.

**Comment by markusdregi** [comment]
I support this 👍🏻

**Comment by sondreso** [comment]
Since no opposing feedback was given to this, I have create an issue for implementing it and will lock the conversation.
